lib/Service/ScheduledMessageService.php

```diff
@@ -197,4 +201,27 @@ public function getScheduledMessageCount(Room $chat, Participant $participant):
 			$participant->getAttendee()->getActorId(),
 		);
 	}
+
+	/**
+	 * @return ScheduledMessage[]
+	 */
+	public function getDue(\DateTime $getDateTime): array {
+		return $this->scheduledMessageMapper->getMessagesDue($getDateTime);
+	}
+
+	public function markAsFailed(ScheduledMessage $message): ScheduledMessage {
+		$metaData = $message->getDecodedMetaData();
+		if (!isset($metaData[ScheduledMessage::METADATA_SEND_AT])) {
+			$metaData[ScheduledMessage::METADATA_SEND_AT] = $message->getSendAt()?->getTimestamp();
+			$message->setMetaData($metaData);
+		}
+		$message->setSendAt(null);
+		$this->scheduledMessageMapper->update($message);
+
+		return $message;
+	}
+
+	public function bulkDelete(array $toDelete): void {
+		$this->scheduledMessageMapper->bulkDelete($toDelete);
```

Comment:
lib/Service/ScheduledMessageService.php:225:34: UndefinedMethod: Method OCA\Talk\Model\ScheduledMessageMapper::bulkDelete does not exist (see https://psalm.dev/022)